### PR TITLE
1864 - Update sync flow and FHIR SDK dependency 

### PR DESCRIPTION
**Pull request by qaziabubakar-vd**
Update sdk dependency

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[https://github.com/opensrp/fhircore/issues/1864] 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/di/CoreModule.kt [hunk 1]
```diff
@@ -39,7 +40,7 @@ import org.smartregister.fhircore.engine.util.helper.TransformSupportServices
 class CoreModule {
   @Singleton
   @Provides
-  fun provideSyncJob(@ApplicationContext context: Context) = Sync.basicSyncJob(context)
+  fun provideSyncJob(@ApplicationContext context: Context) = Sync.oneTimeSync<FhirSyncWorker>(context)
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/di/CoreModule.kt:43]: Can we use the subclass extending the `FhirSyncWorker` instead?

Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/di/CoreModule.kt:43]: Should be this one [`AppSyncWorker`](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/sync/AppSyncWorker.kt)

File: android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt [hunk 2]
```diff
@@ -49,13 +47,12 @@ class SyncBroadcaster
...
 constructor(
   val configurationRegistry: ConfigurationRegistry,
-  val syncJob: SyncJob,
   val fhirEngine: FhirEngine,
   val syncListenerManager: SyncListenerManager,
   val dispatcherProvider: DispatcherProvider
 ) {
 
-  fun runSync() {
+  /*fun runSync() {
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt:55]: Delete commented code/add relevant TODO comment.

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt [hunk 3]
```diff
@@ -91,8 +99,13 @@ constructor(
   val registerRepository: RegisterRepository,
   val dispatcherProvider: DispatcherProvider,
   val workManager: WorkManager,
-  val fhirCarePlanGenerator: FhirCarePlanGenerator
-) : ViewModel() {
+  val fhirCarePlanGenerator: FhirCarePlanGenerator,
+  @ApplicationContext val context: Context,
+  ) : ViewModel() {
+
+  private val _pollState = MutableSharedFlow<SyncJobStatus>()
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:106]:
```suggestion
  private val _syncPollState = MutableSharedFlow<SyncJobStatus>()
```

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt [hunk 4]
```diff
@@ -91,8 +99,13 @@ constructor(
...
+  val pollState: Flow<SyncJobStatus>
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:107]:
```suggestion
  val syncPollState: Flow<SyncJobStatus>
```

File: android/engine/src/main/java/org/smartregister/fhircore/engine/sync/OnSyncListener.kt [hunk 5]
```diff
@@ -16,13 +16,13 @@
...
 
 /**
  * An interface the exposes a callback method [onSync] which accepts an application level FHIR Sync
  * [State].
  */
 interface OnSyncListener {
   /** Callback method invoked to handle sync [state] */
-  fun onSync(state: State)
+  fun onSync(state: SyncJobStatus)
 }
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/sync/OnSyncListener.kt:28]: Do we still need this interface?

**Review by ellykits — CHANGES_REQUESTED** [review]

File: android/engine/build.gradle [hunk 6]
```diff
@@ -211,7 +211,7 @@ dependencies {
         exclude group: 'com.google.android.fhir', module: 'engine'
     }
 
-    api('org.smartregister:engine:0.1.0-beta02-preview6-SNAPSHOT') {
+    api('org.smartregister:engine:0.1.0-beta02-preview8-SNAPSHOT') {
         transitive = true
         exclude group: 'com.google.android.fhir', module: 'common'
```
Comment (ndegwamartin) [inline android/engine/build.gradle:216]: Does the branch break if you add the new SDC artefact as well?

File: android/engine/src/main/java/org/smartregister/fhircore/engine/sync/OnSyncListener.kt (hunk 5 above)
Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/sync/OnSyncListener.kt:28]: I think we still do as per our discussion. Revert the `SyncBroadcaster` code.

**Comment by ellykits** [comment]
Merging this to resolve issues with sync causing app crashes on stage.

**Pull request merged**